pkg/didcomm/protocol/issuecredential/states_test.go

```diff
@@ -711,6 +714,94 @@ func TestCredentialReceived_ExecuteInbound(t *testing.T) {
 
 		require.NoError(t, action(messenger))
 	})
+
+	t.Run("Success (credential with a proof)", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		store := storageMocks.NewMockStore(ctrl)
+		store.EXPECT().Get(gomock.Any()).Return(nil, storage.ErrDataNotFound)
+		store.EXPECT().Put(gomock.Any(), gomock.Any()).Return(nil).Times(2)
+
+		storeProvider := storageMocks.NewMockProvider(ctrl)
+		storeProvider.EXPECT().OpenStore(gomock.Any()).Return(store, nil)
+
+		provider := issuecredentialMocks.NewMockProvider(ctrl)
+		provider.EXPECT().StorageProvider().Return(storeProvider)
+
+		vStore, err := storeVerifiable.New(provider)
+		require.NoError(t, err)
+
+		var credential map[string]interface{}
+		// nolint: lll
+		require.NoError(t, json.Unmarshal([]byte(`{
+				"@context": [
+					"https://www.w3.org/2018/credentials/v1",
+					"https://www.w3.org/2018/credentials/examples/v1"
+				],
+				"credentialSubject": {
+					"degree": {
+						"type": "BachelorDegree",
+						"university": "MIT"
+					},
+					"id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+					"name": "Jayden Doe",
+					"spouse": "did:example:c276e12ec21ebfeb1f712ebc6f1"
+				},
+				"expirationDate": "2020-01-01T19:23:24Z",
+				"id": "http://example.edu/credentials/1872",
+				"issuanceDate": "2009-01-01T19:23:24Z",
+				"issuer": {
+					"id": "did:example:76e12ec712ebc6f1c221ebfeb1f",
+					"name": "Example University"
+				},
+				"proof": {
+					"created": "2010-01-01T19:23:24Z",
+					"jws": "eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..lrkhpRH4tWl6KzQKHlcyAwSm8qUTXIMSKmD3QASF_uI5QW8NWLxLebXmnQpIM8H7umhLA6dINSYVowcaPdpwBw",
+					"proofPurpose": "assertionMethod",
+					"type": "Ed25519Signature2018",
+					"verificationMethod": "did:example:123456#key1"
+				},
+				"referenceNumber": 83294849,
+				"type": [
+					"VerifiableCredential",
+					"UniversityDegreeCredential"
+				]
+			}`), &credential))
+
+		registry := vdriMocks.NewMockRegistry(ctrl)
+		registry.EXPECT().Resolve("did:example:123456").Return(&did.Doc{
+			PublicKey: []did.PublicKey{{
+				ID: "#key1",
+				Value: []byte{
+					234, 100, 192, 93, 251, 181, 198, 73, 122, 220, 27, 48, 93, 73, 166,
+					33, 152, 140, 168, 36, 9, 205, 59, 161, 137, 7, 164, 9, 176, 252, 1, 171,
+				},
+			}},
+		}, nil)
+
+		followup, action, err := (&credentialReceived{}).ExecuteInbound(&metaData{
+			verifiable:   vStore,
+			registryVDRI: registry,
+			transitionalPayload: transitionalPayload{
+				Msg: service.NewDIDCommMsgMap(IssueCredential{
+					Type: IssueCredentialMsgType,
+					CredentialsAttach: []decorator.Attachment{
+						{Data: decorator.AttachmentData{JSON: credential}},
+					},
+				}),
+			},
+		})
+
+		require.NoError(t, err)
+		require.Equal(t, &done{}, followup)
+		require.NotNil(t, action)
+
+		messenger := serviceMocks.NewMockMessenger(ctrl)
+		messenger.EXPECT().ReplyTo(gomock.Any(), gomock.Any())
+
+		require.NoError(t, action(messenger))
+	})
```

Comment:
Maybe to add an error case here as well (i.e. corrupt `proof.jws` value and see that we've got an error with some clear description of the cause).

Reply:
I have a test when a credential is invalid  the https://github.com/hyperledger/aries-framework-go/pull/1842/files/4a36c0c1f382b9c8913b52f4e33ed3171921bd19#diff-f9dc1ab6bf3eb1acef404898701c3a3fR639